src/test.js

```diff
@@ -466,20 +461,43 @@ function generateHash( module, testName ) {
 	return hex.slice( -8 );
 }
 
-function synchronize( callback, last ) {
+function synchronize( callback, priority ) {
+	var last = !priority;
+
 	if ( QUnit.objectType( callback ) === "array" ) {
 		while ( callback.length ) {
 			synchronize( callback.shift() );
 		}
 		return;
 	}
-	config.queue.push( callback );
+
+	if ( priority ) {
+		priorityFill( callback );
+	} else {
+		config.queue.push( callback );
+	}
 
 	if ( config.autorun && !config.blocking ) {
 		process( last );
 	}
 }
 
+// Place previously failed tests on a queue priority line, respecting the order they get assigned.
+function priorityFill( callback ) {
```

Comment:
without this *priority line* for failed tests, I would add successive tests that previously failed on the very beggining of the queue.

Let's say with the following tests:

```js
QUnit.test('test 1', function( assert ) {
  assert.ok( true );
});

QUnit.test('test 2', function( assert ) {
  assert.ok( false );
});

QUnit.test('test 3', function( assert ) {
  assert.ok( false );
});
```

The queue would regularly be on the first run:

- `test 1` (ok)
- `test 2` (failed)
- `test 3` (failed)

On the second run, it should be on the following order:

- `test 2` (failed)
- `test 3` (failed)
- `test 1` (ok)

And without this *priority queue* I would have instead:

- `test 3` (failed)
- `test 2` (failed)
- `test 1` (ok)